parsing-scripts/load_initial_data.py

```diff
@@ -0,0 +1,191 @@
+import os, json
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "MacPorts.settings")
+
+import django
+
+django.setup()
+
+from ports.models import Port, Category, Maintainer, Dependency
+
+with open('portindex.json', "r") as file:
+    ports = json.load(file)
+
+# Add All the Categories to the Database using bulk_create
+categories = set()
+for port in ports:
+    try:
+        for category in port['categories']:
+            categories.add(Category(name=category))
+    except KeyError:
+        pass
+batch = list(categories)
+Category.objects.bulk_create(batch)
+
+temp_maintainers = []  #This is used to compare maintainers without querying the database
+for port in ports:
+
+    # Add Ports to the Database One-by-One
+    new_port = Port()
+    try:
+        new_port.portdir = port['portdir']
+    except KeyError:
+        pass
+    try:
+        new_port.variants = port['variants']
+    except KeyError:
+        pass
+    try:
+        new_port.description = port['description']
+    except KeyError:
+        pass
+    try:
+        new_port.homepage = port['homepage']
+    except KeyError:
+        pass
+    try:
+        new_port.epoch = port['epoch']
+    except KeyError:
+        pass
+    try:
+        new_port.platforms = port['platforms']
+    except KeyError:
+        pass
+    try:
+        new_port.long_description = port['long_description']
+    except KeyError:
+        pass
+    try:
+        new_port.version = port['version']
+    except KeyError:
+        pass
+    try:
+        new_port.revision = port['revision']
+    except KeyError:
+        pass
+    try:
+        new_port.closedmaintainer = port['closedmaintainer']
+    except KeyError:
+        pass
+    try:
+        new_port.name = port['name']
+    except KeyError:
+        pass
+    try:
+        new_port.license = port['license']
+    except KeyError:
+        pass
+    try:
+        new_port.installs_libs = port['installs_lib']
+    except KeyError:
+        pass
+    try:
+        new_port.replaced_by = port['replaced_by']
+    except KeyError:
+        pass
+    new_port.save()
+    try:
+        new_port.categories.add(*port['categories'])
+    except KeyError:
+        pass
```

Comment:
I assume you could use one try and one except here?

Reply:
But then if any one of the keys is not present, won't it stop adding others also?

Reply:
OK, sorry, I missed this detail. Now, I would then do it in a slightly different way. There are some keys which are absolutely necessary (port name, port version, port dir, ...), others which are optional (a port may have no variants, no dependencies, no maintainers, no licence [not quite ok, but possible scenario], ...) and some with defaults (if there is no revision or no epoch, the implied value is 0; if the "closedmaintainer" is missing, the value should be false, ...).

Instead of catching an exception while reading a non-existing key, you could instead check if the key exists and act accordingly, avoiding the exception. You should spare exception handling for other strange scenarios like ending up in invalid database entry, too long name etc. 

I would start with important / necessary fields first (port name, ...) and if an error happens there, you should probably print the error and maybe not add the port at all? (If there's a serious error there, I would expect further errors down the road, so it's not as important what happens next.) For other fields you should still catch errors, but doing it once would probably be sufficient. However make sure that you are not reading invalid keys / that you replace them with some default value.

Reply:
Thank You